src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,121 @@
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Tried to create a book but failed to do so");
+            }
+
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                book.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to create a book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Book book = new Book();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_BY_ID_QUERY)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                book = mapResultSetToBook(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to find the book with an id of "
+                    + id, e);
+        }
+        return Optional.of(book);
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_ALL_QUERY)) {
+            List<Book> books = new ArrayList<>();
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book book = mapResultSetToBook(resultSet);
+                books.add(book);
+            }
+            return books;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to find any books", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(UPDATE_QUERY)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Tried to update a book but failed to do so");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to update the book with an id of "
+                    + book.getId(), e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(DELETE_QUERY)) {
+            statement.setLong(1, id);
+
+            int affectedRows = statement.executeUpdate();
+            return affectedRows != 0;
```

Comment:
It's better to check if `affectedRows > 0` to determine if any rows were deleted, as this is more flexible and allows for multiple rows to be deleted if needed.